File: backend/app/schemas/company.py
from datetime import datetime
from typing import List, Optional
from uuid import UUID
from pydantic import BaseModel, ConfigDict, Field


class CompanyIdentifierResponse(BaseModel):
    id: UUID
    company_id: UUID
    identifier_type: str
    identifier_value: str
    source_url: Optional[str] = None
    confidence: Optional[float] = Field(default=None, ge=0, le=1)
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class CompanyBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=255, description="Full registered or trade name")
    official_domain: Optional[str] = Field(default=None, max_length=255)
    description: Optional[str] = None
    industry: Optional[str] = None
    headquarters: Optional[str] = None


class CompanyCreate(CompanyBase):
    pass


class CompanyResponse(CompanyBase):
    id: UUID
    normalized_name: str
    created_at: datetime
    updated_at: datetime
    identifiers: Optional[List[CompanyIdentifierResponse]] = None

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/company.ts
export interface Company {
  id: string;
  name: string;
  normalized_name: string;
  official_domain?: string | null;
  description?: string | null;
  industry?: string | null;
  headquarters?: string | null;
  created_at: string;
  updated_at: string;
}

export interface CompanyIdentifier {
  id: string;
  company_id: string;
  identifier_type: string;
  identifier_value: string;
  source_url?: string | null;
  confidence?: number | null;
  created_at: string;
}
